ceremonies/src/lib.rs

```diff
@@ -1007,6 +1014,42 @@ impl<T: Config> Pallet<T> {
 		Ok(participant_type)
 	}
 
+	/// removes a participant from the registry maintaining continuous indices
+	fn unregister_newbie(
+		cid: CommunityIdentifier,
+		cindex: CeremonyIndexType,
+		participant: &T::AccountId,
+	) -> Result<(), Error<T>> {
+		if <encointer_scheduler::Pallet<T>>::current_phase() != CeremonyPhaseType::REGISTERING {
+			return Err(<Error<T>>::WrongPhaseForUnregistering.into())
+		}
+
+		if !<NewbieIndex<T>>::contains_key((cid, cindex), &participant) {
+			return Ok(())
+		}
+
+		let participant_index = <NewbieIndex<T>>::get((cid, cindex), &participant);
+		let participant_count = <NewbieCount<T>>::get((cid, cindex));
+
+		let last_participant = <NewbieRegistry<T>>::get((cid, cindex), &participant_count)
+			.expect("Indices are continuous, thus index participant_count is Some; qed");
```

Comment:
hmmm, what if participant_count is zero?

Reply:
or: what if the newbie is the only entry? and participant_count is therefore 1?

Reply:
hmmm, gues then it would return above already....

Reply:
i dont fully get why, but i added an explicit check

Reply:
i think we should rely on continuous indices because otherwise if we dont enforece it, the assignment algorithm breaks

Reply:
Yes, but even more important than ensuring functionality is to never panic.